# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/notification/NotificationUtils.h
# Commit: fca3921d09a5
# Full Hash: fca3921d09a59b709a6939f4a1e65f261367e0a6
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2024-11-12 09:32:50
# Regressor Bug: 1928702
# File Overlap Count: 1
# Description:
#   Bug 1928702 - Part 5: Implement RecvShow r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D227690
# ==============================================================================

diff -r 807f1d1ada66 -r fca3921d09a5 dom/notification/NotificationUtils.h
--- a/dom/notification/NotificationUtils.h	Mon Nov 11 20:34:12 2024 +0000
+++ b/dom/notification/NotificationUtils.h	Mon Nov 11 20:34:13 2024 +0000
@@ -19,6 +19,10 @@
 class Document;
 }  // namespace mozilla::dom
 
+namespace mozilla::dom {
+class IPCNotificationOptions;
+}
+
 namespace mozilla::dom::notification {
 
 /**
@@ -66,6 +70,10 @@
 void ComputeAlertName(nsIPrincipal* aPrincipal, const nsString& aTag,
                       const nsString& aId, nsString& aResult);
 
+nsresult PersistNotification(nsIPrincipal* aPrincipal, const nsString& aId,
+                             const nsString& aAlertName,
+                             const IPCNotificationOptions& aOptions,
+                             const nsString& aScope);
 nsresult UnpersistNotification(nsIPrincipal* aPrincipal, const nsString& aId);
 
 enum class CloseMode {
